### PR TITLE
Update Android NDK (r25) to match what is used in pre-built Qt libraries

### PR DESCRIPTION
**Pull request by nirvn**
Hoping this can resolve an obscure armv7-only crash that appeared when we updated to Qt 6.6 (e.g. https://opengisch.sentry.io/issues/4561431917/?project=6119013&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=14d&stream_index=5)

### CODE REVIEW

**Comment by nirvn** [comment]
@m-kuhn , I can successfully build locally, do you know what's wrong here?

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for beae4666904bebd09d00c199fd4918a046fca686: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4835-2df47c084f9ed6cf0eea281dce2e457942925806-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4835-2df47c084f9ed6cf0eea281dce2e457942925806-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4835-2df47c084f9ed6cf0eea281dce2e457942925806-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4835-2df47c084f9ed6cf0eea281dce2e457942925806-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4835-2df47c084f9ed6cf0eea281dce2e457942925806-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4835-2df47c084f9ed6cf0eea281dce2e457942925806-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4835-2df47c084f9ed6cf0eea281dce2e457942925806-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4835-2df47c084f9ed6cf0eea281dce2e457942925806-x86-android.apk)
</details>

**Review by m-kuhn — APPROVED** [review]

File: cmake/Platform.cmake [hunk 1]
```diff
@@ -20,8 +20,9 @@ elseif(VCPKG_TARGET_TRIPLET STREQUAL "x86-android")
 endif()
 
 if(ANDROID_ABI)
-  set(ANDROID_NDK_VERSION "$ENV{ANDROID_NDK_HOME}" CACHE STRING "Android NDK version")
-  set(ANDROID_BUILD_TOOLS_VERSION "30.0.3" CACHE STRING "Android build-tools version")
+  set(ANDROID_NDK_VERSION "$ENV{ANDROID_NDK_VERSION}" CACHE STRING "Android NDK version")
+  set(ANDROID_NDK_HOME "$ENV{ANDROID_NDK_HOME}" CACHE STRING "Android NDK home path")
```
Comment (m-kuhn) [inline cmake/Platform.cmake:24]: Do we need this?

Comment (nirvn) [reply cmake/Platform.cmake:24]: I don't think so, but not sure we should get rid of it. ATM this change merely fix what was clearly a wrong value.

Comment (m-kuhn) [reply cmake/Platform.cmake:24]: I think If we don't need them we should get rid of them.

Comment (nirvn) [reply cmake/Platform.cmake:24]: @m-kuhn , :fearful: :wink:

**Pull request merged**